crates/composer_api/src/lib.rs

```diff
@@ -12,8 +12,57 @@ use std::{
 
 pub const DEFAULT_SERVER_ADDRESS: &str = "localhost:8888";
 
+/// Composer expects `EventMessage` as the incoming probe data.
 #[derive(Serialize, Deserialize, Debug)]
-pub enum Event {
+pub struct EventMessage {
+    /// List of events a probe collected during a specific time window. For a probe generating
+    /// high-frequency events e.g. more than a hundred per second, it's recommended to buffer and
+    /// pack multiple events into a `EventMessage` to avoid overflowing the socket and reduce
+    /// packet overhead.
+    pub events: Vec<Event>,
+}
+
+impl EventMessage {
+    pub fn new(events: Vec<Event>) -> Self {
+        Self { events }
+    }
+
+    pub fn with_event(event: Event) -> Self {
+        let events = vec![event];
+        Self { events }
+    }
+}
+
+impl Default for EventMessage {
+    fn default() -> Self {
+        let events = Vec::default();
+        Self { events }
+    }
+}
```

Comment:
Vec implements default https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Default-for-Vec%3CT,+Global%3E so we should be able to derive it in turn? I also wonder that clippy didn't catch this.